qendpoint-store/src/main/java/com/the_qa_company/qendpoint/store/EndpointTripleSource.java

```diff
@@ -95,68 +96,22 @@ public CloseableIteration<? extends Statement> getStatements(StatementOrder stat
 		boolean graph = endpoint.getHdt().getDictionary().supportGraphs();
 
 		// convert uris into ids if needed
-		Resource newSubj;
-		IRI newPred;
-		Value newObj;
-		Resource[] newContextes;
+
 		long subjectID = this.endpoint.getHdtConverter().subjectToID(subj);
 		long predicateID = this.endpoint.getHdtConverter().predicateToID(pred);
 		long objectID = this.endpoint.getHdtConverter().objectToID(obj);
 		long[] graphID;
 
-		if (subjectID == 0 || subjectID == -1) {
-			newSubj = subj;
-		} else {
-			newSubj = this.endpoint.getHdtConverter().subjectIdToIRI(subjectID);
-		}
-		if (predicateID == 0 || predicateID == -1) {
-			newPred = pred;
-		} else {
-			newPred = this.endpoint.getHdtConverter().predicateIdToIRI(predicateID);
-		}
-		if (objectID == 0 || objectID == -1) {
-			newObj = obj;
-		} else {
-			newObj = this.endpoint.getHdtConverter().objectIdToIRI(objectID);
-		}
-
 		if (graph) {
 			graphID = new long[contexts.length];
-			newContextes = this.endpoint.getHdtConverter().graphIdToIRI(contexts, graphID);
```

Comment:
If the NativeStore isn't in use then we don't need these variables. So it's simplest if we can delay creating them until we are sure we need them.